book/src/going-further/world-insides.md

```diff
@@ -102,10 +102,10 @@ ids: [
 ]
 deleted: Some((0, 0))
 ```
-Finally, we delete `entity1`.  `ids[newest].index` becomes `1` and we have a linked list where we start at the oldest index `0`.  `0` is not the newest index, so we know its value `1` is the index of the next deleted [`EntityId`](https://docs.rs/shipyard/latest/shipyard/struct.EntityId.html).  `1` is the newest index, so we know we have reached the end of the linked list of deleted entries.
+Finally, we delete `entity1`.  `ids[newest].index` becomes `1` and we have a linked list where we start at the oldest index `0`.  `0` is not the newest index, so we know value `1` is the index of the next deleted [`EntityId`](https://docs.rs/shipyard/latest/shipyard/struct.EntityId.html).  `1` is the newest index, so we know we have reached the end of the linked list of deleted entries.
 ```
 ids: [
-    { index: 1, version: 1 },
+    { index: 0, version: 1 },
```

Comment:
It's 1 for sure =)
```suggestion
    { index: 1, version: 1 },
```

Reply:
Ok. I obviously still don't understand how the indexing and versioning works :D.

As far as I understood, `entity0` has been deleted and it's version has been updated to `1` while setting `deleted` to `(0, 0)` as it's the only deleted entity.

After deleting `entity1`, `entity1`'s version is updated to 1 and `deleted` is updated to `(1, 0)`, because `1` is now the newest deleted entity. But why is `entity0`'s index now updated to `1`.

Another think I don't quite understand yet, is how a simple `deleted` tuple can help detecting deleted entities across versions. What happens, if we delete `entity2` instead of `entity1`. 

I would love to get more information on how the internal entity handling works, since I'm kind of struggling to understand it right now :D